src/util/Fetcher.ts

```diff
@@ -30,11 +30,18 @@ export interface IFetcher {
 }
 
 export default class Fetcher implements IFetcher {
+  private fetcher: typeof _fetch;
+
+  constructor() {
+    if (typeof window !== "undefined" && typeof window.fetch === "function") {
+      this.fetcher = window.fetch;
+    } else {
+      this.fetcher = _fetch;
+    }
+  }
+
   async fetch(url: RequestInfo | URL, init?: RequestInit): Promise<Response> {
```

Comment:
I still find it slightly weird that we decide here to offer other parameters than the regular `fetch` method; I don't think it's our task to also accept `URL`. From a drop-in replacement, you'd expect that it works no differently from the original.

Reply:
Yeah - I'm no expert in this area, but I agree the method signature should match standard fetch() **_exactly_** (if possible): https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch. Is there a specific reason to diverge?